ast/desugar/Desugar.cc

```diff
@@ -706,6 +706,38 @@ class DuplicateHashKeyCheck {
     }
 };
 
+// Flattens the key/value pairs from the Kwargs Hash into the destination container.
+// If Kwargs Hash contains any splats, we skip the flattening and append the hash as-is.
+template <typename Container> void flattenKwargs(unique_ptr<parser::Hash> kwargsHash, Container &destination) {
```

Comment:
This was a big pretty chunk of the duplication.

We drop from 12 levels of indentation (!!!) to "only" 8.